kernel/kernel-api/src/main/java/io/delta/kernel/internal/tablefeatures/TableFeatures.java

```diff
@@ -495,6 +500,51 @@ public static Optional<Tuple2<Protocol, Set<TableFeature>>> autoUpgradeProtocolB
     }
   }
 
+  /**
+   * Extracts features overrides from Metadata properties and returns an updated metadata if any
+   * overrides are present.
+   *
+   * <p>Overrides are specified using a key in th form {@linkplain
+   * #PROPERTIES_FEATURE_OVERRIDE_PREFIX} + {featureName}. (e.g. {@code
+   * delta.feature.icebergWriterCompatV1}). The value should be "true" to add the feature.
+   * Currently, removing values is not handled.
+   *
+   * @return A set of features that had overrides and Metadata object with the properties removed if
+   *     any overrides were present.
+   * @throws KernelException if the feature name for the override is invalid or the value is not
+   *     equal to "true".
+   */
+  public static Tuple2<Set<TableFeature>, Optional<Metadata>> extractFeaturePropertyOverrides(
+      Metadata currentMetadata) {
+    Set<TableFeature> features = new HashSet<>();
+    Map<String, String> properties = currentMetadata.getConfiguration();
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      if (entry.getKey().startsWith(PROPERTIES_FEATURE_OVERRIDE_PREFIX)) {
+        String featureName = entry.getKey().substring(PROPERTIES_FEATURE_OVERRIDE_PREFIX.length());
+
+        TableFeature feature = getTableFeature(featureName);
+        features.add(feature);
+        if (!entry.getValue().equals("true")) {
```

Comment:
To confirm -- this is what delta-spark does? uses `true` and not `supported`?

Reply:
I didn't look in depth.  There is certainly code where delta-spark will accept "supported" I believe.  But from the first round of reviews it sounded like we want to use "true".  It seems we can always relax this in the future?

Reply:
https://github.com/delta-io/delta/blob/1517688f7047a623aee3152261b898783f9596dc/spark/src/main/scala/org/apache/spark/sql/delta/actions/TableFeatureSupport.scala#L501

Sorry to have misled you :( I think it is `supported`

Maybe worth quickly askind Michelle? Should we stick with what delta-spark does and leak the `supported` abstract, or should we just use `true`?

Btw -- i didn't mean to encourage you to go and do this refactor early -- apologies for that. Let's get alignment on this before we merge.